modules/solvers/voltage-divider.stanza

```diff
@@ -91,46 +97,61 @@ defn get-resistance-values (goal-r: Double, r-tol: Double, current-tol: Double)
+  val lo-tol = tol(0.0, tolerance)
+  val hi-tol = tol(0.0, tolerance)
+  val lo-dr = tol(0.0)
+  val hi-dr = tol(0.0)
   qsort{loss, _} $
     for rh in hi-res seq-cat :
+      val rh-tol = tol(rh)
       for rl in lo-res seq? :
+        val rl-tol = tol(rl)
         val vo = compute-output-voltage-range(v-in,
-                                              rl,
-                                              rh,
-                                              (- tol), tol, (- tol), tol,
-                                              0., 0., 0., 0.)
+                                              rl-tol,
+                                              rh-tol,
+                                              lo-tol,
+                                              hi-tol,
+                                              lo-dr,
+                                              hi-dr)
         if in-range?(v-out, vo) :
           val loss = abs(typ-value(v-out) - typ-value(vo))
           One(Ratio(rh, rl, loss))
         else :
           None()
 
-public defn compute-output-voltage-range (v-in: Toleranced,
-                                          r-lo-val, r-hi-val,
-                                          min-lo-tol, max-lo-tol, min-hi-tol, max-hi-tol,
-                                          min-lo-dr, max-lo-dr, min-hi-dr, max-hi-dr) -> Toleranced :
-  val vo-min = min-value(v-in) * (r-lo-val * (1.0 + min-lo-dr)) * (1.0 + min-lo-tol)
-                     / ( (r-lo-val * (1.0 + min-lo-dr)) * (1.0 + min-lo-tol)
-                      + (r-hi-val  * (1.0 + max-hi-dr)) * (1.0 + max-hi-tol))
-  val vo-max = max-value(v-in) * (r-lo-val * (1.0 + max-lo-dr)) * (1.0 + max-lo-tol)
-                     / ( (r-lo-val * (1.0 + max-lo-dr)) * (1.0 + max-lo-tol)
-                     + (r-hi-val  * (1.0 + min-hi-dr)) * (1.0 + min-hi-tol))
-  val vo-nom = typ-value(v-in) * (r-lo-val) / ( (r-lo-val) + (r-hi-val) )
-
+public defn compute-output-voltage-range (v-in:     Toleranced,
+                                          r-lo-val: Toleranced, r-hi-val: Toleranced,
+                                          lo-tol:   Toleranced, hi-tol:   Toleranced,
+                                          lo-dr:    Toleranced, hi-dr:    Toleranced) -> Toleranced :
+  val r-lo = r-lo-val * (1.0 + lo-dr) + (1.0 * lo-tol)
+  val r-hi = r-hi-val * (1.0 + hi-dr) + (1.0 * hi-tol)
+
+  val vo-min = (v-in-min * r-lo-min / (r-lo-min + r-hi-val-max)) where :
+    val v-in-min = min-value(v-in)
+    val r-lo-min = min-value(r-lo)
+    val r-hi-val-max = max-value(r-hi)
+  val vo-nom = (v-in-typ * r-lo-typ / (r-lo-typ + r-hi-val-typ)) where :
+    val v-in-typ = typ-value(v-in)
+    val r-lo-typ = typ-value(r-lo)
+    val r-hi-val-typ = typ-value(r-hi)
+  val vo-max = (v-in-max * r-lo-max / (r-lo-max + r-hi-val-min)) where :
+    val v-in-max = max-value(v-in)
+    val r-lo-max = max-value(r-lo)
+    val r-hi-val-min = min-value(r-hi)
+  
   min-typ-max(vo-min, vo-nom, vo-max)
```

Comment:
Ideally this could be written:
```
v-in * r-lo / (r-lo + r-hi)
```
However we're extracting max-value of `r-hi` when other values are lower, and vice versa.